client/src/pages/LyricsView/LyricsDisplay.tsx

```diff
@@ -17,22 +17,34 @@ import "./LyricsDisplay.css";
 
 interface IProps {
   lyrics?: string;
+  syncedLyricsArray?: Array<any>;
   lyricsArtist?: string;
   lyricsTitle?: string;
   geniusUrl?: string;
+  progressMs?: number;
 }
 
 const LyricsDisplay: React.FunctionComponent<IProps> = ({
+  syncedLyricsArray,
   lyrics,
   lyricsArtist,
   lyricsTitle,
-  geniusUrl
+  geniusUrl,
+  progressMs
 }) => {
+  const clonedLyricsArray = syncedLyricsArray === undefined ? [] : [...syncedLyricsArray];
   const classes = useStyles();
   const lyricsRef = useRef<HTMLDivElement | null>(null);
   const searchInputRef = useRef<HTMLInputElement | null>(null);
   const [search, setSearch] = useState("");
   const [searchShown, setSearchShown] = useState(false);
+  const [syncedLyrics, setSyncedLyrics] = useState({
+    array: clonedLyricsArray,
+    before: "",
+    highlighted: "",
+    after: lyrics
+  });
+  const [naturalSongProgress, setNaturalSongProgress] = useState(0);
```

Comment:
Just a thought - would it be possible to derive `syncedLyrics` purely off `progressMs` without the need of keeping it in the state? It would remove the need for two of the useEffects below

Reply:
I am not sure I followed this one yet, I will leave it for last.

Reply:
That sounds fine, towards the end I can give an example of what I mean